tracer/test/Datadog.Trace.Tests/ManualInstrumentation/DuckTypingTests.cs

```diff
@@ -0,0 +1,139 @@
+// </copyright>
+
+extern alias DatadogTraceManual;
+
+using System;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Ci;
+using Datadog.Trace.Ci.Tagging;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.ManualInstrumentation.Ci;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.ManualInstrumentation.Ci.Proxies;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Sampling;
+using Datadog.Trace.TestHelpers;
+using FluentAssertions;
+using Moq;
+using Xunit;
+using ManualBenchmarkDiscreteStats = DatadogTraceManual::Datadog.Trace.Ci.BenchmarkDiscreteStats;
+using ManualBenchmarkHostInfo = DatadogTraceManual::Datadog.Trace.Ci.BenchmarkHostInfo;
+using ManualBenchmarkJobInfo = DatadogTraceManual::Datadog.Trace.Ci.BenchmarkJobInfo;
+using ManualIScope = DatadogTraceManual::Datadog.Trace.IScope;
+using ManualISpan = DatadogTraceManual::Datadog.Trace.ISpan;
+using ManualISpanContext = DatadogTraceManual::Datadog.Trace.ISpanContext;
+using ManualITest = DatadogTraceManual::Datadog.Trace.Ci.ITest;
+using ManualITestSession = DatadogTraceManual::Datadog.Trace.Ci.ITestSession;
+using ManualTestParameters = DatadogTraceManual::Datadog.Trace.Ci.TestParameters;
+using TestStatus = DatadogTraceManual::Datadog.Trace.Ci.TestStatus;
+
+namespace Datadog.Trace.Tests.ManualInstrumentation;
+
+[Collection(nameof(TracerInstanceTestCollection))]
+[TracerRestorer]
+public class DuckTypingTests
+{
+    private const string Skip = "We can't test these as-is because we rely on automatic instrumentation for ducktyping. To test these, import the DuckTyping folder into Datadog.Trace.Manual";
+    private readonly AsyncLocalScopeManager _scopeManager = new();
+    private readonly TracerSettings _settings = new() { StartupDiagnosticLogEnabled = false };
+    private readonly Tracer _tracer;
+
+    public DuckTypingTests()
+    {
+        _tracer = new Tracer(_settings, new Mock<IAgentWriter>().Object, new Mock<ITraceSampler>().Object, scopeManager: _scopeManager, statsd: null);
+    }
+
+    [Fact]
+    public void CanDuckTypeScopeAsManualIScope()
+    {
+        var scope = _tracer.StartActiveInternal("manual");
+        var span = scope.Span;
+        var spanContext = span.Context;
+
+        var manualScope = scope.DuckCast<ManualIScope>();
+
+        // call all the properties to check for duck typing issues
+        var manualSpan = manualScope.Span.Should().BeAssignableTo<ManualISpan>().Subject;
+        manualSpan.SpanId.Should().Be(span.SpanId);
+        manualSpan.SetException(new Exception("MyException"));
+        manualSpan.Error.Should().Be(span.Error).And.BeTrue();
+        manualSpan.Type.Should().Be(span.Type);
+        manualSpan.OperationName.Should().Be(span.OperationName);
+        manualSpan.ResourceName.Should().Be(span.ResourceName);
+        manualSpan.ServiceName.Should().Be(span.ServiceName);
+        manualSpan.TraceId.Should().Be(span.TraceId);
+        manualSpan.SetTag("Test", "SomeValue");
```

Comment:
Maybe also check the return value of SetTag?

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/5235/commits/cf170ae0d62456a036d9d263240d9b1edf93ac5e